.github/workflows/ci.yaml

```diff
@@ -194,3 +193,43 @@ jobs:
         with:
           files: "trento-*"
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+  build-and-push-image:
+    needs: build
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'release' || github.event_name == 'workflow_dispatch'
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+      IMAGE_TAG: "${{ github.event_name == 'release' && github.event.release.tag_name || 'rolling' }}"
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```

Comment:
This needed to add the capability to use an external cache

Reply:
Dude are you _really_ commenting your own PR? :joy:

Reply:
aha yes i thought it would help reviewing since it is a magic `yaml`